<launch>
    <!-- Load Robot model -->
    <param name="robot_description"
        command="$(find xacro)/xacro --inorder '$(find omni_gazebo)/urdf/omni_robot.xacro'" />
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"
        respawn="false" output="screen" />
        
    <!-- Create the Omni Fake Robot -->
    <node pkg="omni_fake" name="omni_fake_node" type="omni_fake_node" output="screen">
        <param name="init_pose_x"       type="double"   value="2.0" />
        <param name="init_pose_y"       type="double"   value="3.0" />
        <param name="init_pose_theta"   type="double"   value="0.0" />
    </node>
    <node pkg="omni_fake" name="odom_path_node" type="odom_path_node">
        <param name="length_path"       type="int"      value="-1" />
    </node>

    <!-- Load RViz config -->
    <node name="rviz" pkg="rviz" type="rviz" required="true" output="screen"
        args="-d $(find omni_fake)/rviz/rviz_config.rviz" />
</launch>